File: c/sum-of-multiples/src/sum_of_multiples.c
#include "sum_of_multiples.h"

unsigned int sum(const unsigned int *factors, const size_t number_of_factors, const unsigned int limit) {
    unsigned int total = 0;

    for (unsigned int n = 1; n < limit; n++) {
        for (size_t factor = 0; factor < number_of_factors; factor++) {
            if (factors[factor] > 0 && n % factors[factor] == 0) {
                total += n;
                break;
            }
        }
    }

    return total;
}

File: go/sum-of-multiples/sum_of_multiples.go
package sumofmultiples

func SumMultiples(limit int, divisors ...int) int {
	sum := 0

	for n := 1; n < limit; n++ {
		for _, divisor := range divisors {
			if divisor > 0 && n%divisor == 0 {
				sum += n
				break
			}
		}
	}

	return sum
}
